annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java

```diff
@@ -0,0 +1,354 @@
+		searchByBodyValue(VALUE_TARGET_LINK, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.LINKING.name().toLowerCase()));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getValues());
+		assertTrue(target.getValues().contains("http://data.europeana.eu/item/123/xyz"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createSemanticLink() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		
+		String requestBody = getJsonStringInput(LINK_SEMANTIC);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK_SEMANTIC, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.LINKING.name().toLowerCase()));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createGraph() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		
+		String requestBody = getJsonStringInput(LINK_SEMANTIC);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RESOURCE_URI, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RELATION, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK_SEMANTIC, TOTAL_BY_ID_FOUND);
+		
+		// validate fields
+		assertTrue(storedAnno.getMotivation().equals(MotivationTypes.LINKING.name().toLowerCase()));
+		assertEquals(storedAnno.getBody().getInternalType(), BodyInternalTypes.GRAPH.name());
+		Graph graphBody = ((GraphBody) storedAnno.getBody()).getGraph();
+		assertNotNull(graphBody.getNode());
+		assertTrue(graphBody.getNode().getHttpUri().equals("http://thesession.org/tunes/52"));
+		assertTrue(graphBody.getRelationName().equals("isSimilarTo"));
+		assertTrue(graphBody.getResourceUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createSemanticTagSpecific() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```

Comment:
this is a search not a create test, please rename

Reply:
changed to search...